### ISSUE TITLE
feat: auto generate swagger decorators

### ISSUE THREAD

**Issue by techsavvyash** [SamagraX-Stencil/stencil#164]

**Comment by Savio629** [comment]
https://github.com/user-attachments/assets/544c6412-29c7-4b93-890c-ecc98063c507

**Comment by techsavvyash** [comment]
@Savio629 the decorators don't seem to have types on them, this won't generate the swagger documentation properly on it's own, without types these decorators don't add much value.

**Comment by Savio629** [comment]
Yeah actual working on it as well.
Forgot to leave a comment.

Regarding the types, I'll have to retrieve it from the methods itself right?
```  
  @ApiOperation({ summary: 'Create an item' })
  @ApiBody({ type: CreateBook1Dto, description: 'Item data' })
  @ApiResponse({ status: 201, description: 'Item created', type: Book1 })
  @Post()
  async create(@Body() book1Dto: CreateBook1Dto): Promise<Book1> {
    return this.book1Service.create(book1Dto);
  }
```
Right now I'm generating the swagger decorators from a basic template
![image](https://github.com/user-attachments/assets/18fb25d3-c183-40c9-92fb-0968c552b06d)

**Comment by techsavvyash** [comment]
yes you would have to get them from the function parameter and return types.
Make sure to handle the cases:
- [ ] 1. Some of the endpoints in the controller already have swagger decorators
    - 1.1 Those decorators don't have proper types just decorators 
- [x] 2. Functions don't have proper type annotations -- default to any in this case
- [ ] 3. There are more than one controller of each HTTP method type -- then each controller is annotated
- [ ] 4. Solve for QueryParams in case of requests
- [ ] 5. Solve for path params in case of dynamic endpoints like `/user/:id`
